imod_coupler/drivers/ribametamod/ribametamod.py

```diff
@@ -296,14 +297,19 @@ def couple(self) -> None:
                     modribmsw_arrays["rib_sprinkling_demand"] = self.ribasim_user_demand
                     n_users = np.shape(self.mapping.msw2rib["sw_sprinkling"])[0]
                     n_priorities = np.size(self.ribasim_user_demand) // n_users
-                    self.ribasim_user_demand.resize(n_priorities, n_users)
-                    self.ribasim_user_demand0 = self.ribasim_user_demand.copy().reshape(
-                        n_priorities, n_users
-                    )
-                    self.ribasim_user_demand0[self.ribasim_user_demand0 > 0] = 1.0
-                    too_many = (
-                        np.where(self.ribasim_user_demand0.sum(axis=0) > 1)[0] + 1
+                    self.ribasim_user_demand.resize(n_users,n_priorities)
+                    self.ribasim_userprio = self.ribasim_user_demand.copy().reshape(
+                        n_users, n_priorities
                     )
+                    self.ribasim_userprio[self.ribasim_userprio > 1] = 1.0
+                    # number of priorities non-zero elements for each NON-MASKED user, i.e. metaswap sprinkling
+                    nprio_per_target = (1-self.mapping.msw2rib['sw_sprinkling_mask']) \
```

Comment:
As discussed: Please change the dims used for the sprinkling mapping to nusers instead of nbasins.